lib/ami_spec/aws_security_group.rb

```diff
@@ -0,0 +1,68 @@
+require 'aws-sdk-ec2'
+require 'forwardable'
+require 'securerandom'
+
+module AmiSpec
+  class AwsSecurityGroup
+    extend Forwardable
+
+    def self.create(**args)
+      new(**args).tap(&:create)
+    end
+
+    def initialize(ec2: Aws::EC2::Resource.new,
+                   group_name_prefix: "ami-spec-",
+                   connection_port: 22,
+                   vpc_id: nil,
+                   subnet_id: nil,
```

Comment:
This command will fail if this is nil. Is nil a reasonable default?

Reply:
You're right. This isn't very clear. Technically, it'll work with either a VPC id or a subnet id. At least one must be provided. However, AmiSpec only ever provides a subnet id.

I'll remove the `vpc_id` parameter, and make the `subnet_id` mandatory.